newrelic/core/agent_protocol.py

```diff
@@ -516,3 +516,94 @@ def connect(
         # can be modified later
         settings.aws_lambda_metadata = aws_lambda_metadata
         return cls(settings, client_cls=client_cls)
+
+
+class OtlpProtocol(AgentProtocol):
+    def __init__(self, settings, host=None, client_cls=ApplicationModeClient):
+        self.HOST_MAP = {
+            "collector.newrelic.com": "otlp.nr-data.net",
+            "collector.eu.newrelic.com": "otlp.eu01.nr-data.net",
+            "gov-collector.newrelic.com": "gov-otlp.nr-data.net",
+            "staging-collector.newrelic.com": "staging-otlp.nr-data.net",
+            "staging-collector.eu.newrelic.com": "staging-otlp.eu01.nr-data.net",
+            "staging-gov-collector.newrelic.com": "staging-gov-otlp.nr-data.net",
+            "fake-collector.newrelic.com": "fake-otlp.nr-data.net",
+        }
+
+        if settings.audit_log_file:
+            audit_log_fp = open(settings.audit_log_file, "a")
+        else:
+            audit_log_fp = None
+
+        otlp_host = self.HOST_MAP.get(host or settings.host, None)
+        if not otlp_host:
+            default = self.HOST_MAP["collector.newrelic.com"]
+            _logger.warn("Unable to find corresponding OTLP host using default %s" % default)
+            otlp_host = default
```

Comment:
I feel like this does warrant a setting so customers could use a sidecar collector if they want.